pkgs/applications/editors/rstudio/default.nix

```diff
@@ -1,115 +1,183 @@
-{ lib, mkDerivation, fetchurl, fetchpatch, fetchFromGitHub, makeDesktopItem, cmake, boost, zlib
-, openssl, R, qtbase, qtxmlpatterns, qtsensors, qtwebengine, qtwebchannel
-, libuuid, hunspellDicts, unzip, ant, jdk, gnumake, makeWrapper, pandoc
+{ lib
+, mkDerivation
+, fetchurl
+, fetchpatch
+, fetchFromGitHub
+, makeDesktopItem
+, cmake
+, boost
+, zlib
+, openssl
+, R
+, qtbase
+, qtxmlpatterns
+, qtsensors
+, qtwebengine
+, qtwebchannel
+, libuuid
+, hunspellDicts
+, unzip
+, ant
+, jdk
+, gnumake
+, makeWrapper
+, pandoc
 , llvmPackages
+, libyamlcpp
+, soci
+, postgresql
+, nodejs
+, mkYarnPackage
+, qmake
 }:
 
-with lib;
 let
-  verMajor = "1";
-  verMinor = "2";
-  verPatch = "5042";
-  version = "${verMajor}.${verMinor}.${verPatch}";
-  ginVer = "2.1.2";
-  gwtVer = "2.8.1";
+  mathJaxSrc = fetchurl {
+    url = "https://s3.amazonaws.com/rstudio-buildtools/mathjax-27.zip";
+    sha256 = "sha256-xWy6psTOA8H8uusrXqPDEtL7diajYCVHcMvLiPsgQXY=";
+  };
+
+  rsconnectSrc = fetchFromGitHub {
+    owner = "rstudio";
+    repo = "rsconnect";
+    rev = "f5854bb71464f6e3017da9855f058fe3d5b32efd";
+    sha256 = "sha256-ULyWdSgGPSAwMt0t4QPuzeUE6Bo6IJh+5BMgW1bFN+Y=";
+  };
 in
 mkDerivation rec {
   pname = "RStudio";
-  inherit version;
-
-  nativeBuildInputs = [ cmake unzip ant jdk makeWrapper pandoc ];
-
-  buildInputs = [ boost zlib openssl R qtbase qtxmlpatterns qtsensors
-                  qtwebengine qtwebchannel libuuid ];
+  RSTUDIO_VERSION_MAJOR = "1";
+  RSTUDIO_VERSION_MINOR = "4";
+  RSTUDIO_VERSION_PATCH = "1717";
+  version = "${RSTUDIO_VERSION_MAJOR}.${RSTUDIO_VERSION_MINOR}.${RSTUDIO_VERSION_PATCH}";
 
   src = fetchFromGitHub {
     owner = "rstudio";
     repo = "rstudio";
     rev = "v${version}";
-    sha256 = "1n67fa357v51j3z1ma8v2ydfsx3y8n10k2svmfcf4mdzsi8w0kc5";
+    sha256 = "sha256-9c1bNsf8kJjpcZ2cMV/pPNtXQkFOntX29a1cdnXpllE=";
   };
 
+  nativeBuildInputs = [
+    cmake
```

Comment:
qmake and cmake? You are sure?

Reply:
https://src.fedoraproject.org/rpms/rstudio/blob/rawhide/f/rstudio.spec#_69
https://src.fedoraproject.org/rpms/rstudio/blob/rawhide/f/rstudio.spec#_193

without qmake 
```
CMake Error at src/cpp/desktop/CMakeLists.txt:96 (message):
  Did not find supported Qt SDK
```